### DISCUSSION TITLE
Multisite language switch redirects back to homepage.

### DISCUSSION THREAD

**Discussion by japgroevemaker** [statamic/cms#6979] [Troubleshooting] [answered]
Hello,

I'm using Statamic with blade and i'm unable to figure out how to make a language switch which stays on the url you are currently visiting when you make a switch. The switch i have made so far redirects back to the homepage after switching the language.

```blade
@foreach ($sites as $locale)
      <a class="uppercase {{ $site->handle === $locale->handle ? 'active' : '' }}" href="{{ $locale->url }}">
          {{ $locale->name }}
      </a>
      @if (!$loop->last)/@endif
@endforeach
```
Can anyone please help me?

Thanks in advance

**Comment by jasonvarga** [comment, marked as answer]
Use the locales tag instead.

https://statamic.dev/tags/locales

```blade
@foreach(Statamic::tag('locales')->id($id) as $item)
    <li><a href="{{ $item['permalink'] }}">{{ $item['locale']['name'] }}</li>
@endforeach
```

**Reply by japgroevemaker** [reply]
Works like a charm! Thanks!
